sdk/src/main/java/io/opentelemetry/sdk/metrics/BaseInstrument.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.metrics.Instrument;
+import io.opentelemetry.metrics.LabelSet;
+import java.util.List;
+import java.util.Map;
+
+abstract class BaseInstrument<B> implements Instrument<B> {
+
+  private final String name;
+  private final String description;
+  private final Map<String, String> constantLabels;
+  private final List<String> labelKeys;
+
+  BaseInstrument(
+      String name, String description, Map<String, String> constantLabels, List<String> labelKeys) {
+    this.name = name;
+    this.description = description;
+    this.constantLabels = constantLabels;
+    this.labelKeys = labelKeys;
+  }
+
+  abstract B create(LabelSet labelSet);
```

Comment:
s/create/createBound?

Reply:
how about `createBoundInstrument`, so we don't go around on that confusion again? :)